src/mantine-spotlight/src/SpotlightProvider.tsx

```diff
@@ -42,10 +42,12 @@ export function SpotlightProvider({
 }: SpotlightProviderProps) {
   const timeoutRef = useRef<number>(-1);
   const [query, setQuery] = useState('');
-  const [actions, { registerActions, removeActions, triggerAction }] = useActionsState(
-    initialActions,
-    query
-  );
+  const [actions, { registerActions, updateActions, removeActions, triggerAction }] =
+    useActionsState(initialActions, query);
+
+  useDidUpdate(() => {
```

Comment:
I think it should belong to useActionsState

Reply:
I had that yesterday in my original PR but that caused a Infinity loop with one of the tests see https://github.com/mantinedev/mantine/pull/1085/commits/931583004ae6bce786730a0621243a719519678e